plugins/nf-azure/src/main/nextflow/cloud/azure/batch/AzBatchService.groovy

```diff
@@ -299,9 +299,14 @@ class AzBatchService implements Closeable {
     protected TokenCredential createBatchCredentialsWithManagedIdentity() {
         log.debug '[AZURE BATCH] Creating Azure Batch client using Managed Identity credentials'
 
-        return new ManagedIdentityCredentialBuilder()
-                .clientId(config.managedIdentity().clientId)
-                .build()
+        final clientId = config.managedIdentity().clientId
+        final credential = new ManagedIdentityCredentialBuilder()
+
+        if (clientId)
+            log.debug "[AZURE BATCH] Managed Identity Client ID: ${clientId}"
+            credential.clientId(clientId)
```

Comment:
```suggestion
        if( clientId )
            credential.clientId(clientId)
```

Reply:
well spotted.

Reply:
there's a dangerous tendency to write code without unit tests, tho